homeassistant/components/workday/config_flow.py

```diff
@@ -0,0 +1,292 @@
+    """Handle a config flow for Workday integration."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> WorkdayOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return WorkdayOptionsFlowHandler(config_entry)
+
+    async def async_step_import(self, config: dict[str, Any]) -> FlowResult:
+        """Import a configuration from config.yaml."""
+
+        abort_match = {
+            CONF_COUNTRY: config[CONF_COUNTRY],
+            CONF_EXCLUDES: config[CONF_EXCLUDES],
+            CONF_OFFSET: config[CONF_OFFSET],
+            CONF_WORKDAYS: config[CONF_WORKDAYS],
+            CONF_ADD_HOLIDAYS: config[CONF_ADD_HOLIDAYS],
+            CONF_REMOVE_HOLIDAYS: config[CONF_REMOVE_HOLIDAYS],
+        }
+        self._async_abort_entries_match(abort_match)
+        return await self.async_step_options(user_input=config)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the user initial step."""
+        errors: dict[str, str] = {}
+
+        if user_input is not None:
+            self.data = user_input
+            return await self.async_step_options()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=DATA_SCHEMA_SETUP,
+            errors=errors,
+        )
+
+    async def async_step_options(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle remaining flow."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            combined_input: dict[str, Any] = {**self.data, **user_input}
+            try:
+                await self.hass.async_add_executor_job(
+                    validate_custom_dates, combined_input
+                )
+            except AddDatesError:
+                errors["add_holidays"] = "add_holiday_error"
+            except RemoveDatesError:
+                errors["remove_holidays"] = "remove_holiday_error"
+            except NotImplementedError:
+                self.async_abort(reason="incorrect_province")
+
+            abort_match = {
+                CONF_COUNTRY: combined_input[CONF_COUNTRY],
+                CONF_EXCLUDES: combined_input[CONF_EXCLUDES],
+                CONF_OFFSET: combined_input[CONF_OFFSET],
+                CONF_WORKDAYS: combined_input[CONF_WORKDAYS],
+                CONF_ADD_HOLIDAYS: combined_input[CONF_ADD_HOLIDAYS],
+                CONF_REMOVE_HOLIDAYS: combined_input[CONF_REMOVE_HOLIDAYS],
+            }
+
+            self._async_abort_entries_match(abort_match)
+            if not errors:
+                name = self.data.get(CONF_NAME, user_input.get(CONF_NAME))
+                return self.async_create_entry(
+                    title=name,
+                    data={},
+                    options={**self.data, **user_input},
```

Comment:
Can you not use `combined_input` here?
```suggestion
                    options=combined_input,
```